datadog/table_datadog_host.go

```diff
@@ -0,0 +1,101 @@
+package datadog
+
+import (
+	"context"
+	"fmt"
+
+	datadog "github.com/DataDog/datadog-api-client-go/api/v1/datadog"
+	"github.com/turbot/steampipe-plugin-sdk/v3/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v3/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v3/plugin/transform"
+)
+
+func tableDatadogHost(ctx context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "datadog_host",
+		Description: "A host is any piece of infrastructure that runs an instance of the Datadog Agent such as a bare metal instance or a VM.",
+		// There is no Datadog hosts endpoint for getting a single host so instead we use a filter to perform an exact match
+		Get: &plugin.GetConfig{
+			Hydrate:    listHosts,
+			KeyColumns: plugin.SingleColumn("name"),
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listHosts,
+			KeyColumns: plugin.KeyColumnSlice{
+				{Name: "name", Require: plugin.Optional},
+			},
+		},
+		Columns: []*plugin.Column{
+			// Top columns
+			{Name: "name", Type: proto.ColumnType_STRING, Description: "Name of the host."},
+			{Name: "id", Type: proto.ColumnType_STRING, Description: "ID of the host."},
```

Comment:
Does the API support filtering on the host ID at all?

Reply:
@cbruno10 The API doesn't support filtering on the host ID. As per the docs - `This endpoint allows searching for hosts by name, alias, or tag`